File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
  // No input:
  if (!input.trim()) { return 'Fine. Be that way!'; }

  // Yelling:
  if (input.toUpperCase() === input && input.toLowerCase() !== input) {
    return 'Whoa, chill out!';
  }

  // Questions:
  if (/\?$/.test(input)) {
    return 'Sure.';
  }

  return 'Whatever.';
};

module.exports = Bob;

File: python/bob/bob.py
def hey(what):
    cleaned = what.strip()

    # If you address him without actually saying anything:
    if not cleaned:
        return "Fine. Be that way!"

    # If you yell at him:
    if cleaned.isupper():
        return "Whoa, chill out!"

    # If you ask him a question.
    if cleaned.endswith("?"):
        return "Sure."

    return "Whatever."
